CMakeLists.txt

```diff
@@ -60,9 +60,11 @@ macro(build_libyaml)
     endif()
   endif()
   include(ExternalProject)
-  externalproject_add(libyaml-10c9078
-    URL https://github.com/yaml/libyaml/archive/10c907871f1ccd779c7fccf6b81a62762b5c4e7b.zip
-    URL_MD5 b595f0ab0735c04c7f6ed0a798a3eff7
+  externalproject_add(libyaml-0.2.5
+    GIT_REPOSITORY https://github.com/yaml/libyaml
+    GIT_TAG 0.2.5
+    GIT_CONFIG advice.detachedHead=false
```

Comment:
Note that I switched to git here because the latest tarball that is released is very broken w.r.t. CMake.

Reply:
I'm a bit confused why the tarball is broken but the tag is ok?

I mean we could use https://github.com/yaml/libyaml/archive/refs/tags/0.2.5.zip right?

I don't know if it matters, I just don't know why we need git to get the code.

Reply:
> I'm a bit confused why the tarball is broken but the tag is ok?

I don't really know why.  But if you pull https://github.com/yaml/libyaml/releases/download/0.2.5/yaml-0.2.5.tar.gz , it is missing crucial pieces of the cmake infrastructure (like the file `tests/CMakeLists.txt`).

However, you are correct that the zip file *does* contain all of the necessary files.  So I'll switch back to that, thanks for pointing it out.